TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -206,21 +206,13 @@ public override bool Visit(Evaluate evaluate)
 
         public override bool Visit(Goback goback)
         {
-            List<Node> firstStatementsAfter = goback.Parent.Children.SkipWhile(cc => !(cc is Nodes.Goback)).Skip(1).ToList();
-            if (firstStatementsAfter.Any() && (firstStatementsAfter.All(cc => cc is Nodes.End) == false))
-            {
-                DiagnosticUtils.AddError(firstStatementsAfter[0], "There should be no statement after \"GOBACK\"", MessageCode.Warning);
-            }
+            CheckLastStatement(goback, "GOBACK");
             return true;
         }
 
         public override bool Visit(Stop stop)
         {
-            List<Node> firstStatementsAfter = stop.Parent.Children.SkipWhile(cc => !(cc is Nodes.Stop)).Skip(1).ToList();
-            if (firstStatementsAfter.Any() && (firstStatementsAfter.All(cc => cc is Nodes.End) == false))
-            {
-                DiagnosticUtils.AddError(firstStatementsAfter[0], "There should be no statement after \"STOP RUN\"", MessageCode.Warning);
-            }
+            CheckLastStatement(stop, "STOP RUN");
```

Comment:
This is nitpicking but 'RUN' may or may not be present in the source so passing the string "STOP" instead of "STOP RUN" would be more appropriate. You can change it while rebasing.

Reply:
Good catch. The issue is only for STOP RUN.
You must test that "RUN" is present before calling CheckLastStatement.

Reply:
done
https://github.com/TypeCobolTeam/TypeCobol/pull/1493/commits/8d386fe98725207fbab5d9679a0a8839fa0394b2